src/licensure.rs

```diff
@@ -57,6 +59,17 @@ impl Licensure {
                 }
             }
 
+            let shebang_re: Regex = Regex::new(r"^#!.*\n").unwrap();
+            let full_content: String = content.clone();
+            let shebang_match_opt = shebang_re.find(&full_content);
```

Comment:
It seems to me that we should only search against the first line since if there is a shebang somewhere else in the file, say like someone has embedded a script within their code, we don't want to enable "shebang mode"

Reply:
From my understanding, the regex is applied to the entire file as a single string, so the ^ is only capturing the start of the file, and no additional lines.

As such, I don't think what you're describing is a concern with the current approach.

I put together a couple of test files to be sure though, and you can see the results below.

```
% licensure a.test 
# Copyright 2024. 
This is an example

#! shebang in the middle

The is the end
```

```
% cat b.test 
#! shebang at top

Here is one at the top
```

```
% licensure b.test 
#! shebang at top
# Copyright 2024. 

Here is one at the top
```

Reply:
Thanks for the explanation